product/views.py

```diff
@@ -0,0 +1,182 @@
+        if category_name == 'vitamins':
+            price      = product_SSPs.first().price   
+            is_soldout = bool(product_SSPs.first().stock == 0)
+        else:
+            price      = {product_SSP.size : product_SSP.price for product_SSP in product_SSPs}
+            is_soldout = {product_SSP.size : bool(product_SSP.stock == 0) for product_SSP in product_SSPs}
+
+        context['productPrice']     = price
+        context['isSoldOut']        = is_soldout
+
+        # similar products
+        goals_product        = product.goal.all()
+        for goal_product in goals_product:
+            similar_product_list = [
+                    {
+                        'id'            : similar_product.id,
+                        'title'         : similar_product.name,
+                        'subtitle'      : similar_product.sub_name,
+                        'imageUrl'      : similar_product.image_set.get(is_main=True).image_url,
+                        'healthGoalList': [goal.name for goal in similar_product.goal.all()],
+
+                        }
+                    for similar_product in goal_product.product_set.exclude(id=product_id)]
+
+            context['similarProduct'] = similar_product_list[:2] 
+
+        return JsonResponse({"data": context, "message": "SUCCESS"}, status=200)
+
+#Add 눌렀을 때 
+class ProductToCartView(View):
+    @login_decorator
+    def post(self, request):
+        try:
+            data          = json.loads(request.body)
+            user          = request.user
+            product_id    = data['productId']
+            product_size  = data.get('productSize', None)
+            product_price = data['productPrice']
+
+            product_price = float(product_price)
+
+            # orders insert
+            if not Order.objects.filter(user=user, order_status=OrderStatus.objects.get(name='주문 전')).exists():
+                shipping_cost = SHIPPING_COST if product_price < 20 else 0
+                order         = Order.objects.create(  # order_info
+                    user            = user,
+                    order_number    = datetime.today().strftime("%Y%m%d") + str(randint(1000, 10000)),
+                    order_status_id = OrderStatus.objects.get(name='주문 전'),
+                    sub_total_cost  = product_price,
+                    shipping_cost   = shipping_cost,
+                    total_cost      = product_price + shipping_cost
+                )
+            else:
+                order = Order.objects.get(user=user, order_status=OrderStatus.objects.get(name='주문 전'))
+                order.sub_total_cost = float(order.sub_total_cost) + product_priceㅑ
+                order.shipping_cost  = SHIPPING_COST if order.sub_total_cost < 20 else 0
+                order.total_cost     = float(order.sub_total_cost) + order.shipping_cost
+                order.save()
+            
+            if not ProductStock.objects.filter(product_id=product_id, size=product_size):
```

Comment:
```suggestion
            if not ProductStock.objects.filter(product_id=product_id, size=product_size).exist():
```
조건에 대한 판단을 위해 objects.filter을 사용할 때 .exists()을 붙여줘야 모든 값을 다 가져오지 않고, 존재 여부만 판단 합니다.

Reply:
네~~